cmd/restic/cmd_key_add.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/restic/restic/internal/errors"
+	"github.com/restic/restic/internal/repository"
+	"github.com/spf13/cobra"
+)
+
+var cmdKeyAdd = &cobra.Command{
+	Use:   "add",
+	Short: "Add a new key (password) to the repository; returns the added key ID",
+	Long: `
+The "add" sub-command creates a new key and validates the key. Returns the new key ID.
+
+EXIT STATUS
+===========
+
+Exit status is 0 if the command is successful, and non-zero if there was any error.
+	`,
+	DisableAutoGenTag: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runKeyAdd(cmd.Context(), globalOptions, addKeyOpts)
+	},
+}
+
+type NewPasswordOptions struct {
+	NewPasswordFile string
+}
+
+type AddKeyOptions struct {
+	NewPasswordOptions
+	Username string
+	Hostname string
+}
+
+var addKeyOpts AddKeyOptions
+
+func init() {
+	cmdKey.AddCommand(cmdKeyAdd)
+
+	flags := cmdKeyAdd.Flags()
+	flags.StringVarP(&addKeyOpts.NewPasswordFile, "new-password-file", "", "", "`file` from which to read the new password")
+	flags.StringVarP(&addKeyOpts.Username, "user", "", "", "the username for new key")
+	flags.StringVarP(&addKeyOpts.Hostname, "host", "", "", "the hostname for new key")
+}
+
+func runKeyAdd(ctx context.Context, gopts GlobalOptions, opts AddKeyOptions) error {
```

Comment:
now that we can do so in a clean way, please return an error if any `args` are passed to `key add`. Please also make the corresponding change to the other commands.